tidoop-mr-lib/README.md

```diff
@@ -1,6 +1,240 @@
+
+From the input point of view, an input directory may contain several files. Thus a requirement for tidoop-mr-lib (and Hadoop in general) is all the files contain data about the same source, with common format, common purpose, as if all the data within them was a single data file. By accomplishing this, then an input folder can be seen as an input dataset, independently of the number of files it comprises.
+
+From the output point of view, something similar occurs. The output of a MapReduce job comprises several files, one per each running reducer. In addition, some metadata is added to the folder, such as logs and an epty file whose name describes the returning status (`_SUCCESS` or not). It does not matter: the entire folder can be seen as an output dataset (an input dataset for another job) since all the files generated by the reducers will share the same format and purpose.
+
+[Top](#top)
+
+###<a name="categories"></a>Job categories
+####<a name="transformationcomputation"></a>Transformation and computation jobs
+
+On the one hand there are <i><b>transformation</b></i> jobs, i.e. MapReduce jobs that apply on most of the data that can be managed in a data science project. For instance, there is a transformation job aboult filtering lines in the input data based on a regular expression; another example could be the job allowing for joining two datasets into a single one.
+
+On the other hand there are <i><b>computation</b></i> jobs in charge of returning a value which is function of the input data. Examples of this kind of operation could be retrieving the number of data lines, or performing some aggregation operation on the dataset, such as adding it (obviously, if the dataset contains numerical information). Output value can be appended to an output dataset, either as it is returned by the computation job, either by prepending a key value.
+
+[Top](#top)
+
+####<a name="genericngsi"></a>Generic and NGSI specific jobs
+
+Of special interest are <i><b>NGSI specific</b></i> jobs that are only valid for NGSI-like datasets, such as those created by [Cygnus](https://github.com/telefonicaid/fiware-cygnus). Usually, these jobs have their <i><b>generic</b></i> counterpart, but the specific format of NGSI-like data (a Json document with several custom fields) makes very hard to obtain the desired transformation or computation. For instance, computing the average of a certain entity's attribute will require at least a map job on all the NGSI data lines in order to obtain the desired value, and then the average job itself; nevertheless, there exists a single NGSI job dealing with such a computation. Another example could be filtering by entity type, which is a transformation: by using the generic version, a complex regular expresion must be used not only covering the entity type value itself but also covering the search for the field name containing the entity type; the NGSI version only requires a regular expresion for the entity type.
```

Comment:
JSON is an acronym (JavaScript Object Notation), thus I think it should be always shown as "JSON" in natural language text and not "Json" (as NGSI).